main.go

```diff
@@ -12,6 +24,125 @@ func init() {
 	_ = flag.Set("logtostderr", "true")
 }
 
+func GetCurrentNode(clientset kubernetes.Clientset) (*corev1.Node, error) {
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		return nil, errors.New("Environment variable NODE_NAME is not set")
+	}
+
+	node, err := clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node: %w", err)
+	}
+
+	return node, nil
+}
+
+func UpdateNodeLabels(
+	clientset *kubernetes.Clientset,
+	instanceData *metadata.InstanceData,
+) error {
+	if instanceData == nil {
+		return fmt.Errorf("instance data received from Linode metadata service is nil")
+	}
+
+	node, err := GetCurrentNode(*clientset)
+	if err != nil {
+		return fmt.Errorf("failed to get the node: %w", err)
+	}
+
+	klog.Infof("Updating node labels with Linode instance data: %v", instanceData)
+
+	node.Labels["linode_label"] = instanceData.Label
+	node.Labels["linode_id"] = strconv.Itoa(instanceData.ID)
+	node.Labels["linode_region"] = instanceData.Region
+	node.Labels["linode_type"] = instanceData.Type
+	node.Labels["linode_host"] = instanceData.HostUUID
```

Comment:
we should probably get a better format than this, something like?

```
decorator.linode.com/label
decorator.linode.com/instance-id
decorator.linode.com/region
decorator.linode.com/instance-type
decorator.linode.com/host
```

Reply:
But we haven't set DNS for the domain `decorator.linode.com`? Is it okay to use it here without making the domain point to this repo?

Reply:
ya it's just namespacing things you use... we can call it whatever we want but we especially dont want it to clash with anything else someone installs i their cluster. this is the more formal version of `linode_` as you did, the dns doesnt actually have to work.